### ISSUE TITLE
Model.create huge memory hog for large sets

### ISSUE THREAD

**Issue by siboulet** [Automattic/mongoose#4403]
Using Model.create to create/insert a large set (especially if the model is quite fat) causes huge memory usage as all the models are instantiated and returned by Model.create.

It be great to have options for telling Model.create NOT to return the array of created documents. Perhaps also an option for specifying how many models to instantiate simultaneously.

If I get this right currently the only alternative is to instantiate the models and do a .save() on each of them manually (ie. using async.mapLimit)?

**Comment by vkarpov15** [comment]
Yeah right now there's no good way to specify how many models are instantiated simultaneously, if you want that you can use async.eachLimit or just write a loop in co. Good feature to support in the future though

**Comment by djanowski** [comment]
If you're on the promises bandwagon, you can use [`Bluebird.map`](http://bluebirdjs.com/docs/api/promise.map.html) with its `concurrency` option.

**Comment by vkarpov15** [comment]
Good suggestion

**Comment by vkarpov15** [comment]
Going to close this one. Right now you can call `create(docs, { ordered: true })` to create docs 1x/time to minimize memory overhead. And in modern JS with async/await, it is much easier to control concurrency if you want something more fine-grained like "only insert 10 docs at a time".
